File: app/src/main/java/com/easydictionary/app/core/translate/yandex/Definition.java
package com.easydictionary.app.core.translate.yandex;

import com.google.gson.annotations.SerializedName;

class Definition {

    @SerializedName("text")
    private String text;

    @SerializedName("pos")
    private String pos;

    @SerializedName("ts")
    private String transcription;

    @SerializedName("tr")
    private Translate [] translates;

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append(text).append(" {").append(pos).append("} - ")
                .append("[").append(transcription).append("] - ");
        for (int i = 0; i < translates.length; i++) {
            Translate tr = translates[i];
            sb.append(tr);
            if(i != translates.length-1){
                sb.append(" ");
            }
        }
        return sb.toString();
    }
}

File: app/src/main/kotlin/org/book2words/translate/yandex/Definition.kt
package org.book2words.translate.yandex

import com.google.gson.annotations.SerializedName

private class Definition : org.book2words.translate.core.Definition {

    SerializedName("text")
    private var text: String? = null

    SerializedName("pos")
    private var pos: String? = null

    SerializedName("ts")
    private var transcription: String? = null

    SerializedName("tr")
    private var translates: Array<Translate>? = null

    override fun getText(): String? {
        return text
    }

    override fun getTranscription(): String? {
        return transcription
    }

    override fun getTranslate(): String? {
        val result = StringBuilder()
        translates!!.forEachIndexed { i, translate ->
            result.append(translate.getText())
            if (i < translates!!.size() - 1 ) {
                result.append("; ")
            }
        }
        return result.toString()
    }
}
